lib/sorbet-schema/hash_transformer.rb

```diff
@@ -19,4 +26,30 @@ def deep_stringify_keys(hash)
       result[key.to_s] = value.is_a?(Hash) ? deep_stringify_keys(value) : value
     end
   end
+
+  private
+
+  sig { params(value: T.untyped).returns(T.untyped)}
+  def unpack_value(value)
+    if value.is_a?(Hash)
+      deep_symbolize_keys(value)
+    elsif value.is_a?(Array)
+      value.map { |inner_val| unpack_value(inner_val) }
+    else
+      value
+    end
+  end
+
+  sig { params(value: T.untyped).returns(T.untyped)}
+  def unpack_value_try_serialize(value)
+    if value.is_a?(Hash)
+      deep_symbolize_keys_serialize_values(value)
+    elsif value.is_a?(Array)
+      value.map { |inner_val| unpack_value_try_serialize(inner_val) }
+    elsif value.respond_to?(:serialize)
+      value.serialize
+    else
+      value
+    end
+  end
```

Comment:
suggestion(blocking): I think combining these methods together and using a boolean flag makes sense. I had a hard time identifying the difference in the two methods.

```suggestion
  sig { params(value: T.untyped, should_serialize: T::Boolean).returns(T.untyped)}
  def unpack_value(value, should_serialize: false)
    if value.is_a?(Hash)
      deep_symbolize_keys(value)
    elsif value.is_a?(Array)
      value.map { |inner_val| unpack_value(inner_val) }
    elsif should_serialize && value.respond_to?(:serialize)
      value.serialize
    else
      value
    end
  end
```

Reply:
Done!